src/coreComponents/mesh/generators/ExternalMeshGeneratorBase.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ExternalMeshGeneratorBase.cpp
+ */
+
+#include "ExternalMeshGeneratorBase.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+ExternalMeshGeneratorBase::ExternalMeshGeneratorBase( string const & name,
+                                                      dataRepository::Group * const parent )
+  : MeshGeneratorBase( name, parent )
+{
+  enableLogLevelInput();
+
+  registerWrapper( viewKeyStruct::filePathString(), &m_filePath ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setRestartFlags( RestartFlags::NO_WRITE ).
+    setDescription( "Path to the mesh file" );
+
+  registerWrapper( viewKeyStruct::translateString(), &m_translate ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( { 0.0, 0.0, 0.0 } ).
+    setDescription( "Translate the coordinates of the vertices by a given vector (prior to scaling)" );
+
+  registerWrapper( viewKeyStruct::scaleString(), &m_scale ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( { 1.0, 1.0, 1.0 } ).
+    setDescription( "Scale the coordinates of the vertices by given scale factors (after translation)" );
```

Comment:
I thought about it. I concluded that `InternalMeshGenerator` (and it's derived classes) have their own set of nontrivial geometric transformations that they do (such as bending the mesh into a radial one), and it's not clear to me how translation/scaling should combine with those. Since I can't (or don't want to) implement the proper use of these attributes in `InternalMesh`, I don't want to create a false impression (e.g. in rST docs) that they can be applied there.

I think this goes along with the comment by @jeannepellerin that mesh transformations, if we want them, should be separate from generation/import. I'm not in a position right now to fund the right place for it. Whoever wants to extend this functionality can move these attributes elsewhere later.

Reply:
I had in mind that `translate` and `rotate` could be applied on top the `*MeshGenerator` generated nodes.
So you would only need the node coordinates and apply the geometric transformation "_in place_".
Like
```c++
translateAndRotate( cellBlockManager.nodeCoordinates() );
```
?